File: src/test/resources/dev/hilla/generator/nullability/BasicEndpoint.ts
import client from '../../../../connect-client.default.js';
import type { EndpointRequestInit } from '@hilla/frontend';

const box: { (n: number, init?: EndpointRequestInit): Promise<number | undefined> } = async (n, init) => {
    return client.call('dev.hilla.generator.nullability.BasicEndpoint', 'box', { n }, init);
};

const nonNullListOfNonNullStrings: { (argument: string[], init?: EndpointRequestInit): Promise<void> } = async (argument, init) => {
    return client.call('dev.hilla.generator.nullability.BasicEndpoint', 'nonNullListOfNonNullStrings', { argument }, init);
};

const notNull: { (init?: EndpointRequestInit): Promise<string> } = async (init) => {
    return client.call('dev.hilla.generator.nullability.BasicEndpoint', 'notNull', {}, init);
};

const BasicEndpoint = {
    box,
    nonNullListOfNonNullStrings,
    notNull,
};

export default BasicEndpoint;

File: generator/src/test/java/dev/hilla/generator/nullability/BasicEndpoint.java
package dev.hilla.generator.nullability;

import dev.hilla.parser.annotations.Endpoint;
import dev.hilla.parser.annotations.Nonnull;
import java.util.List;

@Endpoint
public class BasicEndpoint {
  public Integer box(int n) {
    return n;
  }

  @Nonnull
  public String notNull() {
    return "";
  }

  public void nonNullListOfNonNullStrings(@Nonnull List<@Nonnull String> argument) {}
}
